Replace the Fluentd DeploymentConfig and DaemonSet note to EFK upgrade

mburke5678 opened the pull request:
https://github.com/openshift/openshift-docs/issues/21638

Do we need this note in the 3.11 EFK upgrade? 
```

If your Fluentd DeploymentConfig and DaemonSet for the EFK components are already set with: 

image: <image_name>:<vX.Y>         
imagePullPolicy: IfNotPresent    

The latest version <image_name> might not be pulled if there is already one with the same 
<image_name:vX.Y> stored locally on the node where the pod is being re-deployed. If so, manually 
change the DeploymentConfig and DaemonSet to imagePullPolicy: Always to make sure it is re-pulled.
```

Preview: 
https://deploy-preview-30755--osdocs.netlify.app/openshift-enterprise/latest/upgrading/automated_upgrades.html#upgrading-efk-logging-stack-dots
https://deploy-preview-30755--osdocs.netlify.app/openshift-enterprise/latest/upgrading/automated_upgrades.html#upgrading-efk-logging-stack-nodots

mburke5678 commented:
@anpingli Is this an issue you can help with? I removed this note [1] and am not sure why. Do we need this info [2] in the 3.11 upgrade docs?

[1] https://github.com/openshift/openshift-docs/pull/17931/files#diff-20aef49252d49e01d0111069c7458fe548450c7a5461bb5ffe7ad8bde4ff47a7L759

[2] https://docs.openshift.com/container-platform/3.10/upgrading/automated_upgrades.html#upgrading-efk-logging-stack

anpingli commented:
@mburke5678  I think the image version will be overwritten in deploymentconfig and DaemonSet after the upgrade playbook is executed. Let me confirmed the next day.

anpingli commented:
@mburke5678  The test prove we still need this note.

anpingli commented:
Instead of update DeploymentConfig and DaemonSet.  you can set openshift_image_tag=v3.11.<Z> Or oreg_url=registry.access.redhat.com/openshift3/ose-${component}:v3.11.<Z> before run playbook

mburke5678 commented:
@anpingli I replaced the _If so, manually  changed the DeploymentConfig and DaemonSet to imagePullPolicy: Always to make sure it is re-pulled._ sentence and replaced it with _You should set the `openshift_image_tag=v3.11` Ansible variable before upgrading. This ensures that the Fluentd DeploymentConfig and DaemonSet are using the latest image version._. 
Does this meet your expecation?
Will the image in the Fluentd DeploymentConfig and Daemonset always be an openshift image?

anpingli commented:
@mburke5678 I was confused about the new sentence. why we prevent the customer to upgrade logging? if you don't want to upgrade logging, don't run logging playbook.

mburke5678 commented:
@anpingli I updated the sentence to: _[If your image version is v3.11, and you want to upgrade to the latest version using the playbook, set the `openshift_image_tag=v3.11.<Z>` or `oreg_url=registry.access.redhat.com/openshift3/ose-${component}:v3.11.<Z>` Ansible parameter.](https://github.com/openshift/openshift-docs/pull/30755/files#diff-20aef49252d49e01d0111069c7458fe548450c7a5461bb5ffe7ad8bde4ff47a7R888-R890)_ 
Sorry for the confusion.

anpingli reviewed (commented):
/lgtm

mburke5678 commented on upgrading/automated_upgrades.adoc:
```diff
@@ -916,6 +933,44 @@ logging deployment.
+You can set the `openshift_logging_elasticsearch_replace_configmap` parameter to `true` to replace your
+`logging-elasticsearch` ConfigMap  with the current default values. In some cases, using an older
+ConfigMap can cause the upgrade to fail. The default is set to `false`. For more information, see the parameter
+in xref:../install_config/aggregate_logging.adoc#aggregate-logging-ansible-variables[specify logging Ansible variables].
```
I copied these two steps from the following procedure. Not sure why I didn't put them here originally. Review as desired.

mburke5678 commented on upgrading/automated_upgrades.adoc:
```diff
@@ -916,6 +933,44 @@ logging deployment.
 
 If the xref:upgrading-efk-logging-stack-script[script above] indicates your indices do not contain fields with a dot in the name, use the following steps to upgrade.
 
+. Review how to
```
I copied these two steps from the following procedure. Not sure why I didn't put them here originally. Review as desired.

bergerhoffer replied:
I'm fine with these as they are since they are copied from existing.

bergerhoffer commented on upgrading/automated_upgrades.adoc:
```diff
@@ -872,6 +872,23 @@ You can set the `openshift_logging_elasticsearch_replace_configmap` parameter to
++
+[NOTE]
+====
+If your Fluentd `DeploymentConfig` and `DaemonSet` for the EFK components are
```
As you mentioned in slack, I don't think these are marked up in 3.11. I'd consider adding "objects" after the two though.

mburke5678 replied:
@bergerhoffer 
I didn't find any references to _`DeploymentConfig` object_ or _`DaemonSet` object_ in 3.11. I did just find some _`*DeploymentConfig*` object_ and _`*DaemonSet*` objects_. Quite inconsistent are these docs!

bergerhoffer commented on upgrading/automated_upgrades.adoc:
```diff
@@ -872,6 +872,23 @@ You can set the `openshift_logging_elasticsearch_replace_configmap` parameter to
+        imagePullPolicy: IfNotPresent
+----
+
+The latest version <image_name> might not be pulled if there is already one with the same 
```
<image_name> should be in backticks

bergerhoffer commented on upgrading/automated_upgrades.adoc:
```diff
@@ -872,6 +872,23 @@ You can set the `openshift_logging_elasticsearch_replace_configmap` parameter to
+----
+
+The latest version <image_name> might not be pulled if there is already one with the same 
+<image_name:vX.Y> stored locally on the node where the pod is being re-deployed. 
```
This one too in backticks.

bergerhoffer commented on upgrading/automated_upgrades.adoc:
```diff
@@ -916,6 +933,44 @@ logging deployment.
+in xref:../install_config/aggregate_logging.adoc#aggregate-logging-ansible-variables[specify logging Ansible variables].
++
+[NOTE]
+====
```
Same comments for this note as the above note.

bergerhoffer reviewed (commented):
Just a few minor comments

mburke5678 commented:
Travis passed:
https://travis-ci.com/github/openshift/openshift-docs/builds/221660330
